### PR TITLE
Add SparkRapidsBuildInfoEvent to the event log

### LINKED ISSUE #10721: [FEA] Dump the complete set of build-info properties to the Spark eventLog
**Is your feature request related to a problem? Please describe.**
We  log only a subset of properties at the WARN level that is likely to be visible in the log4j output

**Describe the solution you'd like**
Create a SparkEvent with the complete build info spanning all the props stored in rapids-4-spark, spark-rapids-ini, cudf and spark-rapids-private jars.

**Describe alternatives you've considered**
Previously there was no consensus if we can dump it to the main driver/executor logs. Either way the more places this info gets logged to the easier it is to find for debugging

**Additional context**
Internal issues

### PR DESCRIPTION
**Pull request by gerashegalov**
Closes #10721

```json
{
  "Event": "com.nvidia.spark.rapids.SparkRapidsBuildInfoEvent",
  "sparkRapidsBuildInfo": {
    "url": "https://github.com/NVIDIA/spark-rapids",
    "branch": "gerashegalov/issue10721",
    "date": "2024-04-30T04:47:11Z",
    "cudf_version": "24.06.0-SNAPSHOT",
    "user": "gerashegalov",
    "revision": "d321e6b5555b1ad16dc515e70d1082e65172cefd",
    "version": "24.06.0-SNAPSHOT"
  },
  "sparkRapidsJniBuildInfo": {
    "url": "https://github.com/NVIDIA/spark-rapids-jni.git",
    "branch": "HEAD",
    "gpu_architectures": "70;75;80;86;90",
    "date": "2024-04-28T05:18:44Z",
    "user": "root",
    "revision": "f3d5933601860c766a15b34968fa79af21ecaa86",
    "version": "24.06.0-SNAPSHOT"
  },
  "cudfBuildInfo": {
    "url": "https://github.com/rapidsai/cudf.git",
    "branch": "HEAD",
    "gpu_architectures": "70;75;80;86;90",
    "date": "2024-04-28T05:18:41Z",
    "user": "root",
    "revision": "064dd7b02166cc67e882b708d66621bc3fafd70b",
    "version": "24.06.0-SNAPSHOT"
  },
  "sparkRapidsPrivateBuildInfo": {
    "url": "https://gitlab-master.nvidia.com/nvspark/spark-rapids-private.git",
    "branch": "HEAD",
    "date": "2024-04-28T04:40:09Z",
    "user": "",
    "revision": "b8ec681d75ce850404b9340dc39d290ef735b3ec",
    "version": ""
  }
}
```
Signed-off-by: Gera Shegalov <gera@apache.org>

### CODE REVIEW

**Comment by gerashegalov** [comment]
build

**Comment by gerashegalov** [comment]
build

File: sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/TrampolineUtil.scala [hunk 1]
```diff
@@ -217,4 +218,9 @@ object TrampolineUtil {
         1
     }
   }
+
```
Comment (jlowe) [inline sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/TrampolineUtil.scala:221]:
```suggestion
```

**Review by jlowe — APPROVED** [review]
Minor nit, lgtm.

**Pull request merged**